src/sidecar/logging.test.ts

```diff
@@ -0,0 +1,294 @@
+import * as assert from "assert";
```

Comment:
Some of these tests were over in the sidecarManager test suite. Move to here where they should have been, and add in new ones for `divineSidecarStartupFailureReason`.